x-pack/plugins/security_solution/public/detections/components/callouts/callout_switcher.tsx

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { FC, memo } from 'react';
+
+import { CallOutMessage } from './callout_types';
+import { CallOut } from './callout';
+import { useCallOutStorage } from './use_callout_storage';
+
+export interface CallOutSwitcherProps {
+  namespace?: string;
+  condition: boolean;
+  message: CallOutMessage;
+}
+
+const CallOutSwitcherComponent: FC<CallOutSwitcherProps> = ({ namespace, condition, message }) => {
+  const { isVisible, dismiss } = useCallOutStorage([message], namespace);
+
+  const shouldRender = condition && isVisible(message);
+  return shouldRender ? <CallOut message={message} onDismiss={dismiss} /> : null;
+};
+
+export const CallOutSwitcher = memo(CallOutSwitcherComponent);
```

Comment:
What's our approach to memoizing components?

I can see it's used pretty extensively throughout the code. The [docs](https://reactjs.org/docs/react-api.html#reactmemo) say "This method only exists as a performance optimization. Do not rely on it to “prevent” a render, as this can lead to bugs."

Reply:
I believe we typically memoize any exported component whether it be inline or on a new line like you've done here except in certain cases where they might be memoized elsewhere already (like some of the [components in forms](https://github.com/elastic/kibana/blob/master/x-pack/plugins/security_solution/public/detections/components/rules/add_item_form/index.tsx#L59) where the components they're passed to are memoized. Someone with a more in depth knowledge of the entire codebase might be able to give you more edge case examples)

Reply:
Thank you, Davis 👍